File: haskell/06/retirement.hs
import System.IO
import Data.Time

promptInt q = do
    putStr $ q ++ " "
    hFlush stdout
    x <- getLine
    return (read x :: Int)

getThisYear = do
    t <- getCurrentTime
    let (y, _, _) = toGregorian $ utctDay t
    return $ fromInteger y

yearLeftMessage y = "You have " ++ (show y) ++ " years left until you can retire."
retireYearMessage y dy = "It's " ++ (show y) ++ ", so you can retire in " ++ (show $ y + dy) ++ "."

main = do
    currentAge <- promptInt "What is your current age?"
    retireAge <- promptInt "At what age would you like to retire?"
    let yearsLeft = retireAge - currentAge
    y <- getThisYear
    putStrLn $ yearLeftMessage yearsLeft
    putStrLn $ retireYearMessage y yearsLeft

File: python/06/retirement.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
from datetime import date

current_age = int(input("What is your current age? "))
retire_age = int(input("At what age would you like to retire? "))

current_year = date.today().year
retire_year = current_year + retire_age - current_age

print(f"It's {current_year}, so you can retire in {retire_year}")
